CONTRIBUTING.md

```diff
@@ -44,43 +44,82 @@ Also, if we consider this may damage the stack (modifications in the engine or t
 
-We are using *.yml* (and not *.yaml*) for all YAML files.
+### General and architecture
 
-We try to not use tabulations, but instead double space for indentation.
+1. Stack was always aimed to **maximum simplicity**, even at the price of non-elegant ways.
+A naïve code is, I think, better than an advanced one. Consider step by step understanding.
+Remember the smart guy who made the ultimate compressed piped command, that nobody, even this guy, can understand after 2 months.
+We should always consider debugging cost, and think as a non-developer user.
 
-Variables that contains Jinja2 code should be prepend with “j2_”.
+2. Stack was designed to be a ***united*** stack, while focusing on ***modularity***.
+This means each role should be fully autonomous against the others, even in the core.
+However, this also means roles are not expected to be used externally of the stack, without the stack standard inventory.
 
-If you are working on Microsoft Windows, please check you do not submit ^M at the end of your lines.
+3. Stack is desired *semi-automatic*. System administrator must always have a way to force things manually, most of the time using variables precedence mechanism.
+This is the reason why in some key roles, inventory data is used instead of system facts.
 
-## Ownership
+### Documenting, comments
 
-By default, each contributing physical person will be added to a “Contributors” list in the main stack README file. This is important for us.
+1. When possible, try to add balanced comments, considering you are teaching to someone.
+Do not hesitate to add url to your references, or to tutorials, etc.
 
-If you do not wish so, please inform us.
+2. **Always document**, if global in main documentation, if role related in role readme. Documentation is written in sphinx, but if this is an issue for you, use markdown and we will translate it to sphinx for you.
 
-## License
+### Scripting
 
-The stack is under MIT. This choice is based on a wish to allow everyone to use it. Please consider using the same license if you wish to Pull Request some code into the main project.
+1. Scripts should be written in *Python 3.6+*, and as simple as possible. Maximum but balanced comments and verbosity should be considered.
 
-If you do not wish so, but you still wish to contribute, please contact us.
+2. A tool like Ansible is a configuration deployment tool. Ansible job is not to do administration tasks. This is scripts job.
+
+3. Tools, scripts, wrappers, are made to simplify system usage (Shell commands, Ansible, Linux tools, etc.). However, when possible, manual way should always be documented side by side with the automated way, to allow easy debug and simply understand scripts, as the stack also has a teaching objective.
+
+4. If a WebUI is available (and it is planed), manual deployment and usage will remains the default way.
```

Comment:
By manual deployment, do you mean CLI? Deployment is still automated by Ansible.

Reply:
Yes, badly written on my side. I will correct this :-)